contracts/auction/MysteriousTreasure.sol

```diff
@@ -93,23 +94,25 @@ contract MysteriousTreasure is Ownable, AuctionSettingIds {
                 (block.number)
             )));
 
-
-        for(uint i = 0; i < 5; i++) {
-            if (totalBoxNotOpened > 1) {
-                // recources in resourcePool is set by owner
-                // nad totalBoxNotOpened is set by rules
-                // there is no need to consider overflow
-                // goldReward, woodReward, waterReward, fireReward, soilReward
-                resourceRewards[i] = uint16(seed % (2 * resourcePool[i] / totalBoxNotOpened));
-                
-                // update resourcePool
-                _setResourcePool(i, resourcePool[i] - resourceRewards[i]);
-            }
-
-            if(totalBoxNotOpened == 1) {
-                resourceRewards[i] = uint16(resourcePool[i]);
-                _setResourcePool(i, resourcePool[i] - resourceRewards[i]);
-            }
+            for(uint i = 0; i < 5; i++) {
+                if (totalBoxNotOpened > 1) {
+                    // recources in resourcePool is set by owner
+                    // nad totalBoxNotOpened is set by rules
+                    // there is no need to consider overflow
+                    // goldReward, woodReward, waterReward, fireReward, soilReward
+                    uint resourceReward = seed % (2 * resourcePool[i] / totalBoxNotOpened);
+                    // 2 ** 16 - 1
+                    require(resourceReward <= 65535);
```

Comment:
Here is may throw revert() errors, I think do somthing like following is better?
```
resourceReward = resourceReward%65535
```